src/main/Main.hs

```diff
@@ -848,6 +852,16 @@ execCmd ExecOpts {..} go@GlobalOpts{..} = do
                munlockFile lk -- Unlock before transferring control away.
                exec eoEnvSettings cmd args
 
+-- | Evaluate some haskell code inline.
+evalCmd :: ExecOpts -> GlobalOpts -> IO ()
+evalCmd e@ExecOpts { eoArgs = args } go@GlobalOpts {..} =
+    if length args /= 1
+    then error "Expecting ONE string argument to 'eval'"
```

Comment:
Yeah, I get what you're saying.  Parsers are pretty composable but probably not worth the trouble in this case.